include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -738,6 +744,55 @@ class __future : private std::tuple<_Args...>
     }
 };
 
+// Tag __async_mode describe a pattern call mode which should be executed asynchronously
+struct __async_mode
+{
+};
+// Tag __sync_mode describe a pattern call mode which should be executed synchronously
+struct __sync_mode
+{
+};
+// Tag __deferrable_mode describe a pattern call mode which should be executed
+// synchronously/asynchronously : it's depends on ONEDPL_ALLOW_DEFERRED_WAITING macro state
+struct __deferrable_mode
+{
+};
+
+template <typename _Mode>
+struct __wait_future_result;
+
+template <>
+struct __wait_future_result<__async_mode>
+{
+    template <typename _Future>
+    void
+    operator()(_Future& /*__future*/) const
+    {
+    }
+};
```

Comment:
Do we really need to make this dispatch generic, for an "arbitrary" `_Future` type? Or would it make more sense to have that dispatch being really within the actual future type used in the implementation?

Reply:
Something like this, for example (not tested):
```
// within class template __future
template<typename _ModeTag>
void wait(_ModeTag)
{
    if constexpr (std::is_same_v<_ModeTag,__sync_mode>)
        wait();
    else if constexpr (std::is_same_v<_ModeTag,__deferrable_mode>)
        __deferrable_wait();
}
```

Reply:
Fixed